main/settings/src/main/scala/sbt/Reference.scala

```diff
@@ -78,7 +78,7 @@ object Reference {
       case LocalRootProject    => "{<this>}<root>"
       case LocalProject(id)    => "{<this>}" + id
       case RootProject(uri)    => "{" + uri + " }<root>"
-      case ProjectRef(uri, id) => "{" + uri + "}" + id
+      case ProjectRef(uri, id) => s"""ProjectRef(uri("$uri"),"$id")"""
```

Comment:
> The fact the the users never comes across non-current ProjectRef is simply untrue. 
  
Can you provide examples of normal builds that make use of `ProjectRef` and whose use is not source dependencies (like sbt)? My point is that no one uses `ProjectRef` in their builds, they only use reference to project definitions. If you're so sure this is not the case, prove me wrong.
  
> Just start sbt on a fresh machine, you'll see it load your global plugins, and then your meta build. These need to be addressed.
  
When you start up sbt, you do not use `Reference.display` to show the URI from which the project is loaded. A proof:
```
jvican in /data/rw/code/scala/sbt                                     [11:37:24] 
> $ sbt                                                     [±global-caching ▴▾]
[info] Loading settings from plugins.sbt ...
[info] Loading project definition from /data/rw/code/scala/sbt/project
[info] Loading settings from build.sbt ...
[info] Set current project to sbtRoot (in build file:/data/rw/code/scala/sbt/)
[info] sbt server started at 127.0.0.1:5842
> 
```
  
>  You can make builds that stitches together source dependencies with the same project ids.
  
You can do very dangerous things with sbt. It's not that sbt disallows you from doing it. If users define source dependencies with the same project ids, it's their fault. We apply the same logic with the misuse of other sbt features, even though sbt should forbid these use cases and it does not.
 
I understand your motivation not to lose information, but I'm yet to be convinced of a use case that **requires** the users to know where a project is located and that they need to copy-paste into their builds.
  
Also, you've missed to replace the use of `Reference.display` for `conflictWarning`:     `conflictWarning := conflictWarning.value.copy(label = Reference.display(thisProjectRef.value))`.